CMakeLists.txt

```diff
@@ -298,6 +298,29 @@ if(ENABLE_COVERAGE)
     set(COVERAGE_LIBRARY --coverage)
 endif()
 
+# FlatBuffers
+set(FLATBUFFERS_BUILD_FLATLIB ON CACHE BOOL "")
+set(FLATBUFFERS_BUILD_FLATC ON CACHE BOOL "")
+set(FLATBUFFERS_BUILD_FLATHASH OFF CACHE BOOL "")
+set(FLATBUFFERS_BUILD_TESTS OFF CACHE BOOL "")
+add_subdirectory(extlib/flatbuffers
+                 ${CMAKE_CURRENT_BINARY_DIR}/extlib/flatbuffers
+                 EXCLUDE_FROM_ALL)
+
+# Buld Q3D header file from schema
+set(q3d_schema ${CMAKE_CURRENT_SOURCE_DIR}/extlib/q3d/object.fbs)
+set(q3d_outdir ${CMAKE_CURRENT_BINARY_DIR}/extlib/q3d)
+set(q3d_header ${q3d_outdir}/object_generated.h)
+
+add_custom_command(
+    OUTPUT ${q3d_header}
+    COMMAND $<TARGET_FILE:flatc> --cpp --no-includes -o ${q3d_outdir}
+            ${q3d_schema}
+    DEPENDS flatc)
+
+add_custom_target(gen_q3d_header DEPENDS ${q3d_header})
+set(Q3D_INCLUDE_DIR ${q3d_outdir})
```

Comment:
This should go either to the q3d repo (preferable) or to src/ with all the other generated files (second best).

Reply:
I was thinking of providing CMakeLists inside the q3d repo for generating the flatbuffer header, but that would mean it would either provide its own flatbuffers/flatc, or the parent project would have to somehow pass flatc/flatbuffers information to the subproject.

I think I'll move the generation to `src/`, especially if SolveSpace will be using FlatBuffers for other things.

Reply:
> or the parent project would have to somehow pass flatc/flatbuffers information to the subproject.

That's easy. You are already doing this. `$<TARGET_FILE:flatc>` is project-wide.